go-controller/pkg/ovn/controller/admin_network_policy/baseline_admin_network_policy.go

```diff
@@ -0,0 +1,259 @@
+		err = c.createNewANP(desiredBANPState, desiredACLs, desiredPorts, true)
+		if err != nil {
+			return fmt.Errorf("failed to create BANP %s: %v", desiredBANPState.name, err)
+		}
+		// since transact was successful we can finally populate the cache
+		c.banpCache = desiredBANPState
+		return nil
+	}
+	var ops []ovsdb.Operation
+	// BANP state existed in the cache, which means its either a BANP update or pod/namespace add/update/delete
+	klog.V(3).Infof("Baseline Admin network policy %s was found in cache...Syncing it", currentBANPState.name)
+	// Did BANP.Spec.Ingress Change (rule inserts/deletes)? && || Did BANP.Spec.Egress Change (rule inserts/deletes)? && ||
+	// If yes we need to fully recompute the acls present in our BANP's port group; Let's do a full recompute and return.
+	// Reason behind a full recompute: Each rule has precendence based on its position and priority of BANP; if any of that changes
+	// better to delete and recreate ACLs rather than figure out from caches
+	// TODO(tssurya): Investigate if we can be smarter about which ACLs to delete and which to add
+	// rather than always cleaning up everything and recreating them. But this is tricky since rules have precendence
+	// from their ordering.
+	// NOTE: Changes to admin policies should be a rare action (can be improved post user feedback) - usually churn would be around namespaces and pods
+	fullPeerRecompute := (len(currentBANPState.egressRules) != len(desiredBANPState.egressRules) ||
+		len(currentBANPState.ingressRules) != len(desiredBANPState.ingressRules))
+	if fullPeerRecompute {
+		// full recompute
+		// which means update all ACLs and address-sets
+		klog.V(3).Infof("BANP %s with priority (old %d, new %d) was updated", desiredBANPState.name, currentBANPState.anpPriority, desiredBANPState.anpPriority)
+		ops, err = c.constructOpsForRuleChanges(desiredBANPState, true)
+		if err != nil {
+			return fmt.Errorf("failed to create update BANP ops %s: %v", desiredBANPState.name, err)
+		}
+	}
+
+	// Did BANP.Spec.Ingress rules get updated?
+	// (at this stage the length of BANP.Spec.Ingress hasn't changed, so individual rules either got updated at their values or positions are switched)
+	// The fields that we care about for rebuilding ACLs are
+	// (i) `ports` (ii) `actions` (iii) priority for a given rule
+	// The changes to peer labels, peer pod label updates, namespace label updates etc can be inferred
+	// from the podIPs cache we store.
+	// Did the BANP.Spec.Ingress.Peers Change?
+	// 1) BANP.Spec.Ingress.Peers.Namespaces changed && ||
+	// 2) BANP.Spec.Ingress.Peers.Pods changed && ||
+	// 3) A namespace started or stopped matching the peer && ||
+	// 4) A pod started or stopped matching the peer
+	// If yes we need to recompute the IPs present in our BANP's peer's address-sets
+	if !fullPeerRecompute && len(desiredBANPState.ingressRules) == len(currentBANPState.ingressRules) {
```

Comment:
How is that for ANP we do a reflect.DeepEqual and here for BANP we just compare the number of rules?
Doesn't make a lot of senes because if `!fullPeerRecompute` is `true` then `len(desiredBANPState.ingressRules) == len(currentBANPState.ingressRules)` is guaranteed to be true as well.

This might probably be solved with my previous comment.

Reply:
Hey good catch.. this match is wrong... ugh it should be same as anp... 
I will be doing a FUP cleanup of some other things.. will include this

And also sharing more code with banp/anp.....